### ISSUE TITLE
Rewrite the blob-server in golang?

### ISSUE THREAD

**Issue by skx** [skx/sos#3]
This would be a cute exercise, and if the code is clean and simple we can do the same with the upload/download server.

In the latter case we'd probably put the upload/download script in one binary, and add command-line flags like:

```
  --read-from 0.0.0.0
  --upload-from 127.0.0.1,192.168.0.0/24
```

+/- IPv6.

**Comment by skx** [comment]
Happily I wrote an "upload/download" server in the past - so most of the code is within my reach:
- https://github.com/skx/go-experiments/tree/master/publishr

**Comment by skx** [comment]
Implemented in a66ab3560e6990c17a1c3afaa5e3b59723895326
